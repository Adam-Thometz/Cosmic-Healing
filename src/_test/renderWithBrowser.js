import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

function withProvider({ children, initialRoutes }) {
  return (
    <MemoryRouter initialEntries={initialRoutes}>
      {children}
    </MemoryRouter>
)
};

function renderWithBrowser(component, options) {
  const initialRoutes = (options && options.initialRoutes)
    ? options.initialRoutes
    : ['/'];
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      initialRoutes
    }), ...options
  });
};

export default renderWithBrowser;