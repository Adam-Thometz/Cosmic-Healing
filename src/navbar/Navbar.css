.Navbar {
  padding: 8px 32px;
  height: 64px;
  position: relative;
}

.Navbar button {
  position: absolute;
  top: 16px;
  right: 5%;
}

.Navbar-links a {
  color: #333333;
  text-decoration: none;
  font-size: 18px;
  font-family: Nuckle;
  font-weight: 700;
  padding: 12px 20px;
  line-height: 16px;
}

.Navbar-links .active {
  color: #D87B41;
}