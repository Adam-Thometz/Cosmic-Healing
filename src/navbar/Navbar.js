import React from 'react';

import './Navbar.css';

import Button from '../_components/button/Button';

const Navbar = () => {
  return (
    <nav className='Navbar'>
      <img src='' alt='' />
      <section className='Navbar-links'>
        <a tabIndex={0} href='#'>Home</a>
        <a tabIndex={0} href='#aboutMe'>About Me</a>
        <a tabIndex={0} href='#services'>Services</a>
      </section>
      <Button primary>Book a free session</Button>
    </nav>
  );
};

export default Navbar;