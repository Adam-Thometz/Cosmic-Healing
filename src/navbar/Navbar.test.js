import React from "react";
import renderWithBrowser from '../_test/renderWithBrowser';
import Navbar from './Navbar';

describe('Navbar component', () => {
  it('renders without crashing', () => {
    renderWithBrowser(<Navbar />);
  });

  it('matches the snapshot', () => {
    const { asFragment } = renderWithBrowser(<Navbar />);
    expect(asFragment()).toMatchSnapshot();
  });
});