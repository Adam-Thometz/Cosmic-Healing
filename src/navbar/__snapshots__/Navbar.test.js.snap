// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component matches the snapshot 1`] = `
<DocumentFragment>
  <nav
    class="Navbar"
  >
    <img
      alt=""
      src=""
    />
    <section
      class="Navbar-links"
    >
      <a
        href="#"
        tabindex="0"
      >
        Home
      </a>
      <a
        href="#aboutMe"
        tabindex="0"
      >
        About Me
      </a>
      <a
        href="#services"
        tabindex="0"
      >
        Services
      </a>
    </section>
    <button
      class="Button primary"
    >
      Book a free session
    </button>
  </nav>
</DocumentFragment>
`;
